public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -14,50 +23,137 @@ interface BaseAlertmanagerArgs {
   });
 
-  return templatesRequestState;
+  return k8sApiSupported ? k8sTemplatesRequestState : templatesRequestState;
+}
+
+function templateGroupsToTemplates(
+  templateGroups: ComGithubGrafanaGrafanaPkgApisAlertingNotificationsV0Alpha1TemplateGroupList
+): NotificationTemplate[] {
+  return templateGroups.items.map((templateGroup) => templateGroupToTemplate(templateGroup));
+}
+
+function templateGroupToTemplate(
+  templateGroup: ComGithubGrafanaGrafanaPkgApisAlertingNotificationsV0Alpha1TemplateGroup
+): NotificationTemplate {
+  return {
+    // K8s entities should always have a metadata.name property. The type is marked as optional because it's also used in other places
+    uid: templateGroup.metadata.name ?? templateGroup.spec.title,
+    name: templateGroup.spec.title,
+    template: templateGroup.spec.content,
+    provenance: templateGroup.metadata.annotations
+      ? templateGroup.metadata.annotations[PROVENANCE_ANNOTATION]
+      : PROVENANCE_NONE,
+  };
 }
 
 function amConfigToTemplates(config: AlertManagerCortexConfig): NotificationTemplate[] {
   return Object.entries(config.template_files).map(([name, template]) => ({
+    uid: name,
     name,
     template,
     // Undefined, null or empty string should be converted to PROVENANCE_NONE
     provenance: (config.template_file_provenances ?? {})[name] || PROVENANCE_NONE,
   }));
 }
 
+function amConfigToTemplate(config: AlertManagerCortexConfig, name: string): NotificationTemplate | undefined {
+  const templates = amConfigToTemplates(config);
+  return templates.find((t) => t.name === name);
+}
+
+interface GetTemplateParams extends BaseAlertmanagerArgs {
+  uid: string;
+}
+
+export function useGetNotificationTemplate({ alertmanager, uid }: GetTemplateParams) {
+  const { useLazyGetAlertmanagerConfigurationQuery } = alertmanagerApi;
+  const { useLazyReadNamespacedTemplateGroupQuery } = templatesApi;
+
+  const [fetchAmConfig, amConfigStatus] = useLazyGetAlertmanagerConfigurationQuery({
+    selectFromResult: (state) => ({
+      ...state,
+      data: state.data ? amConfigToTemplate(state.data, uid) : undefined,
+      currentData: state.currentData ? amConfigToTemplate(state.currentData, uid) : undefined,
+      // TODO set error and isError in case template is not found
+    }),
+  });
+  const [fetchTemplate, templateStatus] = useLazyReadNamespacedTemplateGroupQuery({
+    selectFromResult: (state) => ({
+      ...state,
+      data: state.data ? templateGroupToTemplate(state.data) : undefined,
+      currentData: state.currentData ? templateGroupToTemplate(state.currentData) : undefined,
+    }),
+  });
+
+  const k8sApiSupported = shouldUseK8sApi(alertmanager);
+
+  useEffect(() => {
+    if (k8sApiSupported) {
+      fetchTemplate({ namespace: getK8sNamespace(), name: uid });
+    } else {
+      fetchAmConfig(alertmanager);
+    }
+  }, [alertmanager, uid, k8sApiSupported, fetchAmConfig, fetchTemplate]);
+
+  return k8sApiSupported ? templateStatus : amConfigStatus;
```

Comment:
What pattern do you prefer - `useEffect`, or `skip`? I can't decide, but I'm probably leaning towards `skip`, but not set on that. (Not necessarily something to change here anyway, as we have a mixture already)

Reply:
I need to check the `useEffect` pattern, because `skip` has some side effects and I need to check if `useEffect` has the same issues

Reply:
I would add a todo here to decide the pattern to follow and be consistent everywhere